common/workunit/workunit.cpp

```diff
@@ -8142,6 +8195,106 @@ bool CLocalWorkUnit::getDebugValueBool(const char * propname, bool defVal) const
     return p->getPropBool(prop.str(), defVal);
 }
 
+StringBuffer & getProcessLogDateFromLogFileName(const char * processType, const char * logFileNameWithPath, StringBuffer & logDate)
+{
+    //WU xml has a 'log' attribute for each log file used by a process. The code here
+    //tries to parse log date from the 'log' attribute if the attribute contains the
+    //log date information. Take a thor master log as an example (ECLAgent log is
+    //similar). The 'log' attribute looks like '...thormaster.yyyy_mm_dd.log' if the
+    //'log' attribute does contain the log date information. If thor master stores all
+    //of its log lines for a WU into one log file, there is no need for the log file
+    //name to have the 'yyyy_mm_dd' part (10 byte long). The log attribute may look
+    //like '...thormaster.log'.
+    //In the following code, the datePtr is pointed at the end of the 'thormaster.'.
+    //If the strlen(datePtr) < 10, it means that the log file name does not have the
+    //'yyyy_mm_dd' part. If the strlen(datePtr) >= 10, we read the 10 bytes from the
+    //datePtr as the logDate.
+    const char * searchStr = nullptr;
+    if (strieq(processType, "Thor"))
+        searchStr = THORMASTERLOGSEARCHSTR;
+    else
+        searchStr = ECLAGENTLOGSEARCHSTR;
+
+    const char * logFileName = pathTail(logFileNameWithPath);
+    const char * datePtr = strstr(logFileName, searchStr);
+    if (!datePtr)
+    {
+        IWARNLOG("The %s not found in %s.", searchStr, logFileNameWithPath);
+        return logDate;
+    }
+
+    datePtr += strlen(searchStr);
+    if (strlen(datePtr) < 10) //10 bytes for yyyy_mm_dd
+        return logDate;
+
+    logDate.append(10, datePtr);
+    return logDate;
+}
+
+void CLocalWorkUnit::getWUProcessLogInfo(const char *processType, const char *processName, IArrayOf<IConstWUProcessLogInfo> & processLogs) const
+{
+    if (isEmptyString(processType))
+    {
+        IWARNLOG("CLocalWorkUnit::getWUProcessLogInfo(): empty process type.");
+        return;
+    }
+
+    Owned<IPropertyTreeIterator> processInstances = getProcesses(processType, isEmptyString(processName) ? nullptr : processName);
+    ForEach (*processInstances)
+    {
+        StringBuffer logSpec;
+        IPropertyTree &proc = processInstances->query();
+        proc.getProp("@log", logSpec);
+
+        const char *processName = proc.queryName();
+        if (isEmptyString(processName) || logSpec.isEmpty())
```

Comment:
minor: any reason to use a const char * for processName, but a StringBuffer for logSpec? If you don't need a StringBuffer, better to use proc.queryProp